### PR TITLE
[REFACTOR] Remove the flying var, adds the flying trait.

### PR DESCRIPTION
**Pull request by Adrer**
## What Does This PR Do
Removes the flying variable from /mob/living, instead adds a trait TRAIT_FLYING to use instead.
Adds a variable inherent_traits to /mob/living, any traits in this list will be added to the mob on Initialize.

## Why It's Good For The Game
Single use high-level variables are less than ideal. Using a trait instead is more flexible.
Currently the inherent_traits var is only being used for mobs that happened to have the flying var, but I plan to expand this to all mobs that currently have traits being added during initialize

## Testing
Jump over a chasm using bhop shoes.
Fly over a bunch of tables as a wizard with a broomstick.
As a carp, fly around a bunch, beat some electrocuted windows without taking damage.
Kill a carp, they no longer fly.

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

NPFC

### CODE REVIEW

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 1]
```diff
@@ -354,7 +354,8 @@
 	icon = initial(icon)
 	icon_state = icon_living
 	density = initial(density)
-	flying = initial(flying)
+	if(TRAIT_FLYING in initial_traits) //CHAP-TODO: Unfuck this
```
Comment (BiancaWilkson) [inline code/modules/mob/living/simple_animal/simple_animal.dm:357]: Do you plan on fixing this TODO in this PR, or gonna save it for when you plan on expanding initial_traits? It's fine matter either way, I just want to know what the scope of this is.

**Review by BiancaWilkson — CHANGES_REQUESTED** [review]
Just a couple small things I noticed. Looks good overall!

File: code/modules/mob/living/simple_animal/simple_animal.dm (hunk 1 above)
Comment (Adrer) [reply code/modules/mob/living/simple_animal/simple_animal.dm]: Remove the comment, was just leftover

File: code/modules/mob/living/simple_animal/hostile/retaliate/fish.dm [hunk 2]
```diff
@@ -24,7 +24,6 @@
 	attack_sound = 'sound/weapons/bite.ogg'
 	speak_emote = list("gnashes")
 	faction = list("carp")
-	flying = TRUE
```
[the remark this answers is not in the document]
Comment (Adrer) [reply code/modules/mob/living/simple_animal/hostile/retaliate/fish.dm:27]: Added flight back for these

File: code/modules/mob/living/simple_animal/friendly/butterfly.dm [hunk 3]
```diff
@@ -16,17 +16,13 @@
...
 	density = FALSE
-	flying = TRUE
 	pass_flags = PASSTABLE | PASSGRILLE | PASSMOB
 	ventcrawler = VENTCRAWLER_ALWAYS
 	mob_size = MOB_SIZE_TINY
 	mob_biotypes = MOB_ORGANIC | MOB_BUG
 	butcher_results = list(/obj/item/food/meat = 0)
 	gold_core_spawnable = FRIENDLY_SPAWN
-
-/mob/living/simple_animal/butterfly/Initialize(mapload) //Not the poor butterfly!
```
[the remark this answers is not in the document]
Comment (Adrer) [reply code/modules/mob/living/simple_animal/friendly/butterfly.dm:27]: No, hence why I removed it.

**Review by lewcc — APPROVED** [review]
LGTM

**Review by BiancaWilkson — APPROVED** [review]

**Pull request merged**